docs/concepts.rst

```diff
@@ -0,0 +1,88 @@
+ToDD Concepts
+================================
+
+Before exploring the details of how ToDD actually works, it's important to first spend some
+time understanding a few fundamental concepts.
+
+Groups
+------
+In order to solve the problems that ToDD was created to solve, it was necessary to consider scale
+from the beginning. In ToDD, testing is performed much more like "cattle" than "pets". As a result,
+tests are not performed with discrete endpoints, such as ToDD agents, but rather with "groups".
```

Comment:
Would it make sense to say "...discrete endpoints, such as individual ToDD agents, but rather with..." ? 

Not sure if that's better or not.